### PR TITLE
Pd 15 migrations

### PR DESCRIPTION
**Pull request by rightway1**
Added User models and migrations;  Some updates to readme and config files;

### CODE REVIEW

File: server/models/postgis/user.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#from server.models.dtos.user_dto import UserDTO, UserFilterDTO, Pagination, UserSearchQuery, UserSearchDTO, \
+#     ListedUser
+from server.models.dtos.user_dto import UserDTO
+from server.models.postgis.lookups import UserRole
+from server.models.postgis.utils import NotFound, timestamp
+
+
+class User(db.Model):
+    """ Describes the user account table """
+    __tablename__ = "dmis_users"
```
Comment (hunt3ri) [inline server/models/postgis/user.py]: I'd probably just go with a table name of users, as the database will be dmis

File: server/models/postgis/user.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    def get_by_username(self, username: str):
+        """ Return the user for the specified username, or None if not found """
+        return User.query.filter_by(username=username).one_or_none()
+
+    def update(self, user_dto: UserDTO):
+        """ Update the user details """
+        self.email_address = user_dto.email_address.lower() if user_dto.email_address else None
+        db.session.commit()
+
+    def set_password(self, hashed_password: str):
```
Comment (hunt3ri) [inline server/models/postgis/user.py]: You shouldn't need separate setters for password and role, they can be set on the model and then saved in a oner

So set_password, set_role can be deleted

File: server/models/postgis/user.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+class User(db.Model):
+    """ Describes the user account table """
+    __tablename__ = "dmis_users"
+
+    user_id = db.Column(db.BigInteger, primary_key=True, index=True)
+    username = db.Column(db.String, unique=True, index=True)
+    role = db.Column(db.Integer, default=0, nullable=False)
+    email_address = db.Column(db.String)
+    password = db.Column(db.String)
+
```
Comment (hunt3ri) [inline server/models/postgis/user.py]: Might be useful to have a date_created column here that defaults to the timestamp

File: server/models/dtos/user_dto.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        UserRole[value.upper()]
+    except KeyError:
+        raise ValidationError(f'Unknown mappingLevel: {value} Valid values are {UserRole.ADMIN.name}, '
+                              f'{UserRole.USER.name}')
+
+
+class UserDTO(Model):
+    """ DTO for User """
+    username = StringType()
+    role = StringType()
```
Comment (hunt3ri) [inline server/models/dtos/user_dto.py]: Need to hook up your validator

File: server/models/postgis/user.py (hunk 1 above)
Comment (rightway1) [inline server/models/postgis/user.py]: I think I'd rather keep the name like this for now to distinguish the app control tables from spatial data tables.  Will rename if I get it using a different schema

File: server/models/postgis/user.py (hunk 3 above)
Comment (rightway1) [inline server/models/postgis/user.py]: Done

File: server/models/postgis/user.py (hunk 2 above)
Comment (rightway1) [inline server/models/postgis/user.py]: Done

File: server/models/dtos/user_dto.py (hunk 4 above)
Comment (rightway1) [inline server/models/dtos/user_dto.py]: Work in progress...  Will be in next PR

**Comment by rightway1** [comment]
Pushed an update with fixes for 2 of these issues

**Pull request merged**